scenarios/aws/ec2/vm_run.go

```diff
@@ -83,8 +85,23 @@ func VMRunWithDocker(ctx *pulumi.Context) error {
 		return err
 	}
 
-	_, installDocker := osWithDockerProvided[vm.OS.Descriptor().Flavor]
-	manager, _, err := docker.NewManager(*env.CommonEnvironment, vm, installDocker)
+	_, isDockerInstalled := osWithDockerProvided[vm.OS.Descriptor().Flavor]
+	manager, setupDockerCmd, err := docker.NewManager(*env.CommonEnvironment, vm, !isDockerInstalled)
+	if err != nil {
+		return err
+	}
+
+	ecrCredsHelperInstall, err := vm.OS.PackageManager().Ensure("amazon-ecr-credential-helper", utils.PulumiDependsOn(setupDockerCmd))
```

Comment:
You will need to do this in `datadog-agent` as well I believe, so perhaps it should be put in a dedicated function.
Also we should perhaps revert the dependencies, you can install the helper before Docker, and set it as `DependsOn()` when creating `docker.NewManager`, so that any subsequent call does not need to depend on `ecrConfigCommand`.